addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -418,6 +443,47 @@ function ThreadFactory({ Entity }) {
             }
         }
 
+        /**
+         * Open a dialog to add channels as followers.
+         */
+        promptAddChannelFollower() {
+            this._promptAddFollower({ mail_invite_follower_channel_only: true });
+        }
+
+        /**
+         * Open a dialog to add partners as followers.
+         */
+        promptAddPartnerFollower() {
+            this._promptAddFollower({ mail_invite_follower_channel_only: false });
+        }
+
+        /**
+         * Refresh followers information from server.
+         */
+        async refreshFollowers() {
+            const [{ message_follower_ids: followerIds }] = await this.env.rpc({
+                model: this.model,
+                method: 'read',
+                args: [this.id, ['message_follower_ids']]
+            });
+            if (followerIds && followerIds.length > 0) {
+                const { followers } = await this.env.rpc({
+                    route: '/mail/read_followers',
+                    params: {
+                        follower_ids: followerIds,
+                        context: {} // empty context to be overridden in session.js with 'allowed_company_ids'
```

Comment:
```suggestion
                        context: {}, // empty context to be overridden in session.js with 'allowed_company_ids'
```
Also why can't session.js just create if if it's empty.

Reply:
No idea, just took the rpc call from the old code

Reply:
Ok to merge with this, but please add a "fixme" in the code with the link to a task, task to create in backlog so we can remove that at some point.

Reply:
https://www.odoo.com/web#id=2243187&action=333&active_id=1519&model=project.task&view_type=form&cids=1&menu_id=4720